packages/system/src/Stream/Stream/partition.ts

```diff
@@ -0,0 +1,31 @@
+import * as E from "../../Either"
+import * as T from "../_internal/effect"
+import type * as M from "../_internal/managed"
+import type { Stream } from "./definitions"
+import { partitionEither_ } from "./partitionEither"
+
+/**
+ * Partition a stream using a predicate. The first stream will contain all element evaluated to true
+ * and the second one will contain all element evaluated to false.
+ * The faster stream may advance by up to buffer elements further than the slower one.
+ */
+export function partition_<R, E, O>(
+  self: Stream<R, E, O>,
+  p: (o: O) => boolean,
+  buffer = 16
+): M.Managed<R, E, readonly [Stream<unknown, E, O>, Stream<unknown, E, O>]> {
```

Comment:
given `partitionEither` has `Managed<R, never, ...>` shouldn't this be like `M.Managed<R, never, readonly [Stream<unknown, E, O>, Stream<unknown, E, O>]>`

Also tuples should be flagged readonly

Reply:
Good catch, and it's actually the inferred type... :/